### PR TITLE
DRAFT: timeouts deselection on GPU deselection

### PR DESCRIPTION
**Pull request by samir-nasibli**
### Description

GPU timeouts deselections
*  Kmeans sparse

After initializing centroids via Kmeans_Init it gets stuck here https://github.com/intel/scikit-learn-intelex/blob/1c813372dac6c8d33b745f3679956339c3d5c23e/onedal/cluster/kmeans.py#L254

So actually the issue on onedal backend side.
---

Checklist to comply with before moving PR from draft:

**PR completeness and readability**

- [ ] I have reviewed my changes thoroughly before submitting this pull request.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [ ] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [ ] I have added a respective label(s) to PR if I have a permission for that.  
- [ ] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [ ] The unit tests pass successfully.
- [ ] I have run it locally and tested the changes extensively.

**Performance**

- [ ] I have measured performance for affected algorithms using [scikit-learn_bench](https://github.com/IntelPython/scikit-learn_bench) and provided at least summary table with measured data, if performance change is expected.
- [ ] I have provided justification why performance has changed or why changes are not expected.

### CODE REVIEW

File: .circleci/run_xpu_tests.py
Comment (samir-nasibli) [inline .circleci/run_xpu_tests.py:1]: Just for debug; will be reverted.

**Comment by samir-nasibli** [comment]
> Looking at the deselections: is this just all sparse tests for kmeans?

Yeah, it seems only Kmeans sparse on GPU affected with timeouts. Expecting to have green CI jobs for GPU conformance validation with this PR

**Comment by samir-nasibli** [comment]
Interesting fact: `sklearnex` own tests for the `KMeans` doesn't cover such cases from the conformance deselected.

**Comment by samir-nasibli** [comment]
Last CI run: http://intel-ci.intel.com/ef76956a-860f-f1fb-ada5-a4bf010d0e2e

**Comment by ethanglaser** [comment]
Should be covered by https://github.com/oneapi-src/oneDAL/pull/2906 (if not feel free to re-open)
